### ISSUE TITLE
Second CA needed although marked as optional in README

### ISSUE THREAD

**Issue by ElektroVirus** [camelop/rust-mtls-example#1]
Hey! I tried this project  but can't get it running. I created the certificates (sections 1-11). The project compiles but it panics:
```
$ cargo run -- server
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/rust-mtls-example server`
thread 'main' panicked at 'error binding to 0.0.0.0:3030: certificate parse error', /home/elias/.cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.2/src/server.rs:534:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Only after creating second CA (section 13):
```
openssl genrsa -out second_ca.key 2048
openssl req -new -x509 -key second_ca.key -out second_ca.crt
```
the server starts running. 

This also fixes the problem, if only a single CA is wanted. However I'm not sure if you want to use different CAs with the client and the server. At least the README could also be updated :​) (that's why I'm not creating a pull request)
```
diff --git a/src/main.rs b/src/main.rs
index ea44dcc..5b4f6d4 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -13,7 +13,7 @@ async fn run_server() {
         .tls()
         .key_path("ca/localhost.key")
         .cert_path("ca/localhost.bundle.crt")
-        .client_auth_required_path("ca/second_ca.crt")
+        .client_auth_required_path("ca/ca.crt")
         .run(([0, 0, 0, 0], 3030))
         .await
 }
```

**Environment**
 - OS: Fedora Linux 36
 - Versions of relevant software and tools: rustc 1.54.0-nightly (657bc0188 2021-05-31)

**Comment by camelop** [comment]
Thanks for your suggestion @ElektroVirus ! It can be confusing indeed and I hope the glitch does not waste too much of your time.

Based on your suggestion, I made a note in readme to reflect this: https://github.com/camelop/rust-mtls-example/commit/0b739dbad2f8a1f746a3a55ff1c340a553ce52b1 Hope this clarifies the thing!

**Comment by ElektroVirus** [comment]
Thank you, @camelop ! 

I got your example running completely after some trying, both with one and with two certificate authorities. However, It was also a bit confusing that the client used in the ```main.rs``` is the second_client, which is in the optional section of README (and the client_0 is not used at all), is this a mistake?

But thanks for this project! It's awesome, especially the instructions for the certificate generation, they help a lot. And also it is nice to have this really simple mTLS-implementation written in Rust, this is a great project to learn how it works!

Cheers, mate :​)

**Comment by camelop** [comment]
Thanks for the information. Yeah I think you've also kindly pointed out the same issue in the first comment. So if you check the latest readme file it now says

https://github.com/camelop/rust-mtls-example/blob/0b739dbad2f8a1f746a3a55ff1c340a553ce52b1/readme.md?plain=1#L26

Hope this resolves the confusion!

**Comment by ElektroVirus** [comment]
Yes I read it, but somehow I didn't understand it completely. I thought about them more like two different things - the other thing refers to the thing with 1 or 2 CAs, the other thing is with one or more clients. That's why I wanted to explicitly tell you how I saw it, as I am a novice in this topic. 

However, I must say that your project excels in demonstrating in a simple manner the implementation of mTLS and the usage (and creation) of certificates.  So thanks for it and have a nice beginning week! :)
